### ISSUE TITLE
OdooModuleServiceImpl.getModule IndexNotReadyException

### ISSUE THREAD

**Issue by muhammedashraf9244** [wt-io-it/odoo-pycharm-plugin#67]
Plugin Version: 0.6.2
IntelliJ Version: PyCharm 2022.3 EAP
IntelliJ Build: PC-223.6160.21

<details><summary>CopyReferenceAction#update@EditorPopup(preload-bgt) (com.intellij.ide.actions.CopyReferenceAction), actionId=CopyReference, text='Copy Reference'</summary>

```
com.intellij.openapi.project.IndexNotReadyException: Please change caller according to com.intellij.openapi.project.IndexNotReadyException documentation
	at com.intellij.openapi.project.IndexNotReadyException.create(IndexNotReadyException.java:67)
	...
	at com.intellij.util.indexing.FileBasedIndexEx.getValues(FileBasedIndexEx.java:139)
	at at.wtioit.intellij.plugins.odoo.index.OdooIndex.getValues(OdooIndex.java:49)
	at at.wtioit.intellij.plugins.odoo.modules.impl.OdooModuleServiceImpl.lambda$getModule$1(OdooModuleServiceImpl.java:63)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:860)
	at at.wtioit.intellij.plugins.odoo.modules.impl.OdooModuleServiceImpl.getModule(OdooModuleServiceImpl.java:61)
	at at.wtioit.intellij.plugins.odoo.pycharm.PyCharmOdooAddonsImportResolver.resolveImportReference(PyCharmOdooAddonsImportResolver.java:49)
	at com.jetbrains.python.psi.resolve.PyResolveImportUtil$foreignResults$1.invoke(PyResolveImportUtil.kt:199)
	at com.jetbrains.python.psi.resolve.PyResolveImportUtil$foreignResults$1.invoke(PyResolveImportUtil.kt:199)
	at kotlin.sequences.TransformingSequence$iterator$1.next(Sequences.kt:210)
	...
	at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:808)
	at com.jetbrains.python.psi.resolve.PyResolveImportUtil.foreignResults(PyResolveImportUtil.kt:201)
	...
	at com.jetbrains.python.psi.impl.references.PyReferenceImpl$CachingResolver.resolve(PyReferenceImpl.java:771)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$resolveWithCaching$1(ResolveCache.java:161)
	...
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:147)
	at com.jetbrains.python.psi.impl.references.PyReferenceImpl.multiResolve(PyReferenceImpl.java:94)
	...
	at com.jetbrains.python.psi.impl.references.PyReferenceImpl.resolve(PyReferenceImpl.java:73)
	at com.intellij.ide.actions.FqnUtil.elementToFqn(FqnUtil.java:43)
	...
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroupAsync$15(ActionUpdater.java:395)
	at io.opentelemetry.context.Context.lambda$wrap$1(Context.java:212)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:244)
	...
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:206)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	...
	at java.base/java.lang.Thread.run(Thread.java:833)
```
</details>

**Comment by ap-wtioit** [comment]
@muhammedashraf9244 thanks for reporting this, when trying to reproduce this (File -> Invalidate Caches, Invalidate and Restart, [CTRL]+[ALT]+[SHIFT]+[C] for copy reference of an `from odoo.addons....` import dependent path) i get the following message:
```
'Copy Reference' is not available while PyCharm is updating indexes
```

Do you remember how you triggerd the Copy Reference action? Or is it now available during indexing for 2022.3 (EAP)?

**Comment by bigbear3001** [comment]
this is fixed in 0.6.10 (pending jetbrains approval for the marketplace) if you are eager to test it you can use the jar file from https://github.com/wt-io-it/odoo-pycharm-plugin/tree/v0.6.10

**Comment by ap-wtioit** [comment]
0.6.10 is available in the marketplace now
